File: target-dotnet/src/SESS.NexaERP.Domain/Masters/Customer.cs
using SESS.NexaERP.Domain.Common;

namespace SESS.NexaERP.Domain.Masters;

public sealed class Customer : AuditableEntity
{
    public string CustomerCode { get; set; } = string.Empty;
    public bool IsCustomerCodeLocked { get; set; }
    public string Name { get; set; } = string.Empty;
    public string LegalCustomerName { get; set; } = string.Empty;
    public string? TradeName { get; set; }
    public string CustomerType { get; set; } = string.Empty;
    public string? GstNumber { get; set; }
    public string? PanNumber { get; set; }
    public string? BillingAddress { get; set; }
    public string? ShippingAddress { get; set; }
    public string? State { get; set; }
    public string? StateCode { get; set; }
    public string Country { get; set; } = "India";
    public string? ContactPerson { get; set; }
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? Industry { get; set; }
    public string? PaymentTerms { get; set; }
    public int? CreditPeriodDays { get; set; }
    public decimal? CreditLimit { get; set; }
    public string PortalOrganizationId { get; set; } = string.Empty;
    public string Status { get; set; } = MasterStatuses.Draft;
    public string ApprovalStatus { get; set; } = MasterApprovalStatuses.Draft;
    public string? ApprovedBy { get; set; }
    public DateTimeOffset? ApprovedAt { get; set; }
    public bool IsActive { get; set; } = true;
}

File: target-dotnet/src/SESS.NexaERP.Web/src/types/customer.ts
// Mirrors SESS.NexaERP.Application.Masters customer contracts (PascalCase wire contract).

export interface CustomerSummary {
  Id: string
  CustomerCode: string
  Name: string
  GstNumber: string | null
  PanNumber: string | null
  PortalOrganizationId: string | null
  Status: string
  ApprovalStatus: string
  IsActive: boolean
  Version: number
  CreditLimit: number | null
}

export interface CustomerDetail {
  Id: string
  CustomerCode: string
  Name: string
  LegalCustomerName: string
  TradeName: string | null
  CustomerType: string
  GstNumber: string | null
  PanNumber: string | null
  BillingAddress: string | null
  ShippingAddress: string | null
  State: string | null
  StateCode: string | null
  Country: string
  ContactPerson: string | null
  Phone: string | null
  Email: string | null
  Industry: string | null
  PaymentTerms: string | null
  CreditPeriodDays: number | null
  CreditLimit: number | null
  PortalOrganizationId: string
  Status: string
  ApprovalStatus: string
  IsActive: boolean
  Version: number
  BankMetadata: unknown | null
  AttachmentMetadataJson: string | null
}

export interface UpsertCustomerRequest {
  CustomerCode: string
  LegalCustomerName: string
  TradeName: string | null
  CustomerType: string
  GstNumber: string | null
  PanNumber: string | null
  BillingAddress: string | null
  ShippingAddress: string | null
  State: string | null
  StateCode: string | null
  Country: string
  ContactPerson: string | null
  Phone: string | null
  Email: string | null
  Industry: string | null
  PaymentTerms: string | null
  CreditPeriodDays: number | null
  CreditLimit: number | null
  PortalOrganizationId: string
  Version: number | null
  BankMetadataJson: string | null
  AttachmentMetadataJson: string | null
}
